src/Cassandra/Policies/DCAwareRoundRobinPolicy.cs

```diff
@@ -117,25 +119,36 @@ public DCAwareRoundRobinPolicy(string localDc, int usedHostsPerRemoteDc)
         public void Initialize(ICluster cluster)
         {
             _cluster = cluster;
+
             //When the pool changes, it should clear the local cache
             _cluster.HostAdded += _ => ClearHosts();
             _cluster.HostRemoved += _ => ClearHosts();
+
+            var availableDcs = _cluster.AllHosts().Select(h => h.Datacenter).Where(dc => dc != null).Distinct().ToList();
```

Comment:
NIT: iteration for `ToList()` can be avoided, as `string.Join()` can take an `IEnumerable<string>` instance.

Reply:
Hmm I added `ToList` to avoid the multiple enumarion of the `IEnumerable` but I guess it doesn't make much sense, all it does is add one more iteration to the already existing two.

Reply:
Actually I'd rather have the `ToList` just to make sure that the enumeration occurs only once.